source/auth/auth.rst

```diff
@@ -1251,172 +1246,514 @@ mechanism
 
-.. code: typescript
+Human Callback
+______________
+Drivers that support the `Human Authentication Flow`_ MUST implement the human
+callback version of the `OIDC Callback`.
 
-  function onRequest(info: IdpServerInfo, params: RequestParameters): IdpServerResponse
+In addition to the information described in the `OIDC Callback` section,
+drivers MUST be able to pass the following information to the callback:
 
-In this example, one of the timeout values would then need to be present on ``RequestParameters``.  ``IdpServerResponse`` is defined as:
+- ``idpInfo``: Information used to authenticate with the IdP.
 
-.. code:: idl
+  - ``issuer``: A URL which describes the Authentication Server. This identifier
+    should be the iss of provided access tokens, and be viable for RFC8414
+    metadata discovery and RFC9207 identification.
+  - ``clientId``: A unique client ID for this OIDC client.
+  - ``requestScopes``: A list of additional scopes to request from IdP.
 
-  IdPServerResponse:
-      description: "The result of a token request"
-      strict: false
-      fields:
-          accessToken:
-              description: "The OIDC access token"
-              type: string
-          expiresInSeconds:
-              description: "The expiration time in seconds from the current time"
-              type: int
-              optional: true
-          refreshToken:
-              description: "The OIDC refresh token"
-              type: str
-              optional: true
+- ``refreshToken``: The refresh token, if applicable, to be used by the callback
+  to request a new token from the issuer.
 
-The token refresh callback must take the same arguments as the request callback, as well as the ``refreshToken`` string given by the ``IdpServerResponse``, and might look like the following::
+In addition to the information described in the `OIDC Callback` section, the
+callback MUST be able to return the following information:
+
+- ``refreshToken``: An optional refresh token that can be used to fetch new
+  access tokens.
+
+An example callback API that supports the human callback might look like:
 
 .. code:: typescript
 
-    function onRefresh(info: IdpServerInfo, params: RefreshParameters): IdpServerResponse
+  interface IdpInfo {
+    issuer: string;
+    clientId: string;
+    requestScopes: Optional<Array<string>>;
+  }
 
-Before calling a callback, the driver MUST acquire a lock unique to the cache key.  The driver MUST ensure that credentials have not changed between when the lock was requested and when it was acquired.  The lock MUST be released
-when the callback call as finished or errored.
-This is because request callbacks may involve human interaction, and refresh
-callbacks could use refresh tokens that can only be used once.
+  interface OIDCCallbackParams {
+      callbackTimeoutMS: int;
+      version: int;
+      idpInfo: Optional<IdpInfo>;
+      refreshToken: Optional<string>;
```

Comment:
Do we enforce this being a string? From the driver's perspective, it can be any opaque value (whatever that looks like in a given language), right?

Reply:
No, it can be any value. I changed the type to `any`.